collectors/nvd/collectors.py

```diff
@@ -153,14 +153,59 @@ def collect(self, cve: Union[str, None] = None) -> str:
             if not flaw:
                 continue
 
-            # check if any changes
+            # we are interested in NIST only
+            flaw_nist_cvss_scores = flaw.cvss_scores.filter(
+                issuer=FlawCVSS.CVSSIssuer.NIST
+            )
+
+            # get the original CVSSv2 and CVSSv3 vectors
+            original_cvss2 = (
+                flaw_nist_cvss_scores.filter(version=FlawCVSS.CVSSVersion.VERSION2)
+                .values_list("vector", flat=True)
+                .first()
+            )
+            original_cvss3 = (
+                flaw_nist_cvss_scores.filter(version=FlawCVSS.CVSSVersion.VERSION3)
+                .values_list("vector", flat=True)
+                .first()
+            )
+
+            # get the new CVSSv2 and CVSSv3 vectors
+            new_cvss2 = item["cvss2"].split("/", 1)[1] if item["cvss2"] else None
+            new_cvss3 = item["cvss3"].split("/", 1)[1] if item["cvss3"] else None
+
+            # check if any changes (via FlawCVSS)
+            if original_cvss2 == new_cvss2 and original_cvss3 == new_cvss3:
+                continue
+
+            # check if any changes (via Flaw, will be deprecated)
             if flaw.nvd_cvss2 == item["cvss2"] and flaw.nvd_cvss3 == item["cvss3"]:
                 continue
 
             desync.append(item["cve"])
 
+            # update CVSSv2 and CVSSv3 if necessary (via FlawCVSS)
+            for original_cvss, new_cvss, version in [
+                (original_cvss2, new_cvss2, FlawCVSS.CVSSVersion.VERSION2),
+                (original_cvss3, new_cvss3, FlawCVSS.CVSSVersion.VERSION3),
+            ]:
+                if original_cvss != new_cvss:
+                    flaw_nist_cvss_scores.filter(version=version).delete()
+
+                    cvss_score = FlawCVSS.objects.create_cvss(
+                        flaw=flaw,
+                        issuer=FlawCVSS.CVSSIssuer.NIST,
+                        version=version,
+                        vector=new_cvss,
+                        acl_write=flaw.acl_write,
+                        acl_read=flaw.acl_read,
+                    )
+                    cvss_score.save()
```

Comment:
why are we deleting the existing record and creating a new one with the same characteristics but the vector? why not just update the existing record with the new vector? I feel like in general a lot of the code here could be simplified, or am I missing something?

Reply:
`create_cvss(...)` performs either update or create, so line 193 with `delete()` is redundant - it will only force `create_cvss(...)` to always create a new cvss instead of updating the existing one if it is sufficient.

I forgot to remove line 193 when I was checking score updating. Thanks for noticing!